.ci/run.sh

```diff
@@ -18,6 +18,9 @@
 
 set -e
 
+# Run unit testing
+sudo -E PATH=$PATH bash -c "make check"
```

Comment:
This should be just `make check` since `.ci/go-test.sh` elevates privs to `root` using `sudo` itself. Unless you drop the `sudo` here, the unit tests only get run once - as `root`, but we need them to run twice (as non-`root` and as `root`) to ensure all the unit test coverage scenarios are exercised.

Reply:
I can do that, but the thing is that we .ci/run.sh script is called with sudo, meaning that calling make check without sudo will end up doing the same thing. I mean that the user is gonna be `root` also in that case, right ?
What do you think we should do ?

Reply:
I'm not really following I'm afraid. But we must run `.ci/go-test.sh` as a non-`root` user to allow it to elevate its privileges using `sudo`.

Reply:
I get that, but I am saying that `.ci/run.sh` is run as root. Don't you think this will prevent us from running `go-test.sh` as non root ?

Reply:
Well, we I'm saying we need to make `.ci/go-test.sh` run as non-root, as it does under Travis to ensure we run the tests correctly ;)

Reply:
So, we currently run `sudo -E PATH=$PATH bash -c ".ci/run.sh"` in the Jenkins script - I guess the questions are:
1) Do we need to run that as sudo?
2) Should we run it maybe as both sudo and no sudo?

The default user is `USER=jenkins` according to the logs...

Reply:
If `.ci/run.sh` is already run as `root`, why have you changed that script to add the `sudo` call?

Reply:
(crossed comments in the night....) so, if we just remove the sudo from the current run.sh line in the Jenkins script - do we see any potential problems anybody? /cc @sboeuf @chavafg

Reply:
I don't think we need to run this as sudo since jenkins user has sudo capabilities, I will give this a try. I don't think we should run this twice from a jenkins perspective, otherwise our CI will take forever, only for unit test purpose.

Reply:
We're only running `go test` twice (it's fast and already handled by `.ci/go-test.sh`).